src/test_c2s_srv.c

```diff
@@ -207,30 +208,34 @@ int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
  recfd:
 
       // If a valid connection request is received, client has connected.
-      // Proceed.  Note the new socket fd - recvsfd- used in the throughput test
-      recvsfd = accept(testOptions->c2ssockfd,
-                       (struct sockaddr *) &cli_addr, &clilen);
-      if (recvsfd > 0) {
-        log_println(6, "accept() for %d completed",
-                    testOptions->child0);
+      // The new connection (c2s_conn) should be used in the throughput test.
+      c2s_conn.socket = accept(testOptions->c2ssockfd,
+                               (struct sockaddr *)&cli_addr, &clilen);
+      if (c2s_conn.socket > 0) {
+        log_println(6, "accept() for %d completed", testOptions->child0);
 
         // log protocol validation indicating client accept
-        procstatusenum = PROCESS_STARTED;
-        proctypeenum = CONNECT_TYPE;
-        protolog_procstatus(testOptions->child0, testids, proctypeenum,
-                            procstatusenum, recvsfd);
-	// To preserve user privacy, make sure that the HTTP header
-	// processing is done prior to the start of packet capture, as many
-	// browsers have headers that uniquely identitfy a single user.
+        protolog_procstatus(testOptions->child0, testids, CONNECT_TYPE,
+                            PROCESS_STARTED, c2s_conn.socket);
+        if (testOptions->connection_flags & TLS_SUPPORT) {
+          c2s_conn.ssl = SSL_new(ctx);
+          if (c2s_conn.ssl == NULL) return -ENOMEM;
+          if (SSL_set_fd(c2s_conn.ssl, c2s_conn.socket) == 0) return -EIO;
+          if (SSL_accept(c2s_conn.ssl) != 1) return -EIO;
```

Comment:
Pulled forward from previous rounds

>Can we refactor this into its own function? We repeat the code elsewhere and it will lessen the bloat we're adding to a 400+ line function.
 
>pboothe added a note 4 days ago
>It happens 3 times - in test_c2s, test_s2c, and web100srv, and the last of those three times is subtly and importantly different from the other two.

>My general bar for "write a function" is "the same code three times", which this doesn't hit.



I don't understand the rationale for three times. Why would you want the same code to repeat more than once? If we realize we need to make a change to the way we initialize SSL connections, we don't want to have to grep for all instances of this code and make the change multiple times. I can't see the web100srv instance yet because it isn't in any of the CLs so far, but is it so different that we can't parameterize it?